### PR TITLE
fix: correct schema for role according to proto

### PR DESCRIPTION
**Pull request by krtkvrm**
- [x] Adding types and definition_level for role as being defined in table

| Column name      | type     | Example                  |
|------------------|----------|--------------------------|
| id <PK>          | string   | org_manager              |
| name             | string   | Org Manager              |
| types            | []string | ["user" + "group"]       |
| namespace | string   | organization             |
| metadata         | json     | {"resource": "firehose"} |

- [x] Adding id while creating the role

### CODE REVIEW

**Comment by krtkvrm** [comment]
The validation is breaking due to re-numbering the fields in proto

File: odpf/shield/v1/shield.proto [hunk 1]
```diff
@@ -394,18 +394,19 @@ message ListGroupsResponse {
 message Role {
   string id = 1;
   string name = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
-  repeated string tags = 3;
-  repeated string actions = 4;
+  repeated string types = 3;
+  string definition_level = 4;
```
Comment (AkarshSatija) [inline odpf/shield/v1/shield.proto:398]: Can you help me understand the use case of `definition_level`?

Comment (rsbh) [reply odpf/shield/v1/shield.proto:398]: It tells the policy namespace in which the role is applied. We can filter roles according to it.
We will also use it in dynamic schema generation.
I think "namespace" will be better key name

Comment (krtkvrm) [reply odpf/shield/v1/shield.proto:398]: yup

**Pull request merged**